include/algorithms/kmeans/kmeans_types.h

```diff
@@ -106,54 +105,29 @@ enum PartialResultId
  */
 enum ResultId
 {
-    centroids,                        /*!< Table containing cluster centroids */
-    assignments,                      /*!< Table containing assignments of observations to particular clusters */
-    objectiveFunction,                /*!< Table containing an objective function value */
-    goalFunction = objectiveFunction, /*!< Table containing an objective function value \DAAL_DEPRECATED */
-    nIterations,                      /*!< Table containing the number of executed iterations */
```

Comment:
1) some problem with static BC: not defined symbols.
2) problem with dynamic BC: user will get silently niterations instead of goal-function,
and **similar potential problem with out of allocated memory requests.** 

How did you solve this problem in test system?
Definitely this 'small' fix(deleting of deprecated field) leads to breaking dynamic BC. (**It requires the approval at least!**) 
And how do we plan to inform users?

Reply:
possible solution for potential segmentation fault problem is 
`lastResultId = nIterations + 1`

Reply:
> Definitely this 'small' fix(deleting of deprecated field) leads to breaking dynamic BC. (It requires the approval at least!)

Why do you think that the dynamic failed? I have definitions of these fields in the inner folder.